README.md

```diff
@@ -1,6 +1,306 @@
+object buckets.
+- For each new Service Instance, a new, uniquely named bucket is created.
+- For each new Service Instance Credential, a Secret is generated with the
+coordinates and credentials of the Service Instance's associated bucket.
+- Deleting a Service Instance destroys the associated bucket.
+- Deleting a Service Instance Credential deletes the secret.
+Nothing is done on the broker side.
+This is because the broker does not perform any actions on Bind, so there is nothing to
+undo.
+
+### Limitations
+
+- Currently, the CNS Object Broker is dependent on GCE.  If run outside of a GCE environment, it will fail to start.  This is because the Broker detects the external IP of the node on which it is run.
+It pairs this IP with the port of the gluster-s3-deployment *Service* to generate the coordinaates returned in the *ServiceInstanceCredential*.  This is not a requirement of brokers in general.
+
+- As it is implemented, the CNS Object Broker must run on the same Kubernetes cluster as the gluster-s3-deployment.  This is because it uses a kube api client to get the gluser-s3-deployment's *Service* from which is parses it's external port. This is not a requirement of brokers in general.
+
+- Auth:  The S3 api implementation (Gluster-Swift) does not enforce any authentication / authorization.
+Each new bucket, regardless of the *Namespace* of its *ServiceInstance*, is accessible and mutable by anyone with the coordinates of the S3 server.
+
+- Flat Bucket Hierarchy:  Gluster-Swift's S3 implementation allows for nested buckets.
+However, we have written the CNS Object Broker utilizing the minio-go S3 client, which has no concept of nested buckets.
+This results in an artificially flat bucket hierarchy.
+
+## Installation
+
+### Dependencies
+- [Google SDK](https://cloud.google.com/sdk/) (`gcloud`): To access GCP via cli.
+- [Kubernetes](https://github.com/kubernetes/kubernetes): to run local K8s cluster
+- [Gluster-Kubernetes Deploy Script](https://github.com/copejon/gk-cluster-deploy): to deploy GCP K8s cluster with Gluster-Kubernetes Object Storage and CNS Object Broker
+- Kubernetes-Incubator [Service-Catalog](https://github.com/kubernetes-incubator/service-catalog): to provide Service-Catalog helm chart
+- Kubernetes [Helm](https://github.com/kubernetes/helm#install): to install Service-Catalog chart
+
+### Assumptions
+- Gk-cluster-deploy was written and tested on Fedora 25 and RHEL 7.
+- A [Google Cloud Platform](https://cloud.google.com/compute/) account.
+
+### Topology
+There are two primary systems that make up this demonstration
+They are the Broker and its colocated CNS Object store.
+It should be noted that the Broker can be implemented to run anywhere
+They fill the role of our micro-service provider
+It is only for the purpose of this demo that we decided deploy the Broker and the CNS Cluster in the same location
+
+
+A second system will be the locally running Kubernetes cluster on which with Service-Catalog is deployed
```

Comment:
_A second system will be the locally running Kubernetes cluster on which with Service-Catalog is..._ Add:
Note: the demo uses a locally run, all-in-one Kubernetes cluster; however, any Kubernetes cluster will work.